k8s/upgrade/src/bin/upgrade-job/helm/upgrade.rs

```diff
@@ -224,6 +238,68 @@ impl HelmUpgrade {
         HelmUpgradeBuilder::default()
     }
 
+    /// Use the HelmReleaseClient's upgrade dry runcommand to validate the upgrade command.
+    pub(crate) async fn dry_run(self) -> Result<()> {
+        let mut args_with_dry_run = self.core_chart_extra_args.clone();
+
+        if let Some(mut args) = args_with_dry_run {
+            args.push("--dry-run".to_string());
+            args_with_dry_run = Some(args);
+        } else {
+            args_with_dry_run = Some(vec!["--dry-run".to_string()]);
+        }
+
+        if self.chart_variant == HelmChart::Core && !self.already_upgraded {
+            // It should be impossible to hit this error.
+            let chart_dir = self
+                .core_chart_dir
+                .ok_or(CoreChartUpgradeNoneChartDir.build())?;
+
+            info!("Validating helm upgrade...");
+
+            let command: &str = "helm";
+            let mut args: Vec<String> = vec_to_strings![
+                "upgrade",
+                self.release_name,
+                chart_dir.to_string_lossy(),
+                "-n",
+                self.client.namespace,
+                "--timeout",
+                "15m"
+            ];
+
+            // Extra args
+            args.extend(
+                args_with_dry_run
+                    .unwrap_or_default()
+                    .iter()
+                    .map(ToString::to_string),
+            );
+
+            debug!(%command, ?args, "Helm upgrade dry-run command");
+            let output =
+                Command::new(command)
+                    .args(args.clone())
+                    .output()
+                    .context(HelmCommand {
+                        command: command.to_string(),
+                        args: args.clone(),
+                    })?;
+
+            let stdout_str = str::from_utf8(output.stdout.as_slice()).context(U8VectorToString)?;
+            debug!(stdout=%stdout_str, "Helm upgrade dry-run command standard output");
```

Comment:
Ah now I see what std_err was...
Should we include std_err in the debug log here as well?

Reply:
Its already there 
```
let stdout_str = str::from_utf8(output.stdout.as_slice()).context(U8VectorToString)?;
            debug!(stdout=%stdout_str, "Helm upgrade dry-run command standard output");
``

Reply:
I mean this debug log only includes the stdout and not the stderr, should we include stderr as well?

Reply:
added